### PR TITLE
Mille4ei code review1

### PR DESCRIPTION
**Pull request by Emiller321**
Description of the Project:
An application that will tell you the sunrise and sunset take place anywhere in the world

What I learned from this code review:
* How to start using location within an application 
* How to use GPS capabilities within an application

### CODE REVIEW

**Comment by Emiller321** [comment]
--Analysis of the program
My analysis of the program is I believe this is a really neat app idea. I have not seen an app that specifically tracks the sunrise and sunset times of anywhere in the world. I think this app is a good app for people who are interested in astronomy and how the sun interacts with Earth

--Was the program available in UC Github on time?
Yes the program was available to me on time and has given me ample time to review

--Is the program documented/commented well enough for you to understand?
Yes the program is well documented and is written well enough for me to understand the concept of the application

--Does the program compile?
Yes the program compiles and is running effectively 

--Rationale behind your changes.
The rationale behind my changes in code, was to get rid of whitespace, have better implementation, and coding conventions in Kotlin Language

--Links to three specific commits that you made to your own group's GitHub repository before the end of sprint deadline.  
https://github.com/lh1207/LocoQuest

**Comment by Emiller321** [comment]
My changes are int he "Revert My Code Review Suggestions" Commit

File: app/src/main/java/com/sundial/v1001/MainViewModel.kt [hunk 1]
```diff
@@ -11,7 +11,7 @@ import kotlinx.coroutines.launch
 class MainViewModel (var twilightService : ITwilightService = TwilightService()) : ViewModel() {
     var twilight : MutableLiveData<List<Twilight>> = MutableLiveData<List<Twilight>>()
 
-    fun fetchData(){
+    internal fun fetchData(){
```
Comment (Emiller321) [inline app/src/main/java/com/sundial/v1001/MainViewModel.kt]: Updated the visibility modifiers per the suggestions here: https://kotlinlang.org/docs/visibility-modifiers.html

File: app/src/main/java/com/sundial/v1001/dao/ITwilightDAO.kt [hunk 2]
```diff
@@ -7,5 +7,5 @@ import retrofit2.http.GET
 interface ITwilightDAO {
 
     @GET("/json")
-    fun getData() : Call<ArrayList<Twilight>>
+    internal fun getData() : Call<ArrayList<Twilight>>
```
Comment (Emiller321) [inline app/src/main/java/com/sundial/v1001/dao/ITwilightDAO.kt]: Updated the visibility modifiers per the suggestions here: https://kotlinlang.org/docs/visibility-modifiers.html

File: app/src/main/java/com/sundial/v1001/service/TwilightService.kt [hunk 3]
```diff
@@ -9,10 +9,10 @@ import kotlinx.coroutines.withContext
 import retrofit2.awaitResponse
 
 interface ITwilightService{
-    suspend fun fetchData() : List<Twilight>?
+    internal fun fetchData(): List<Twilight>?
```
Comment (Emiller321) [inline app/src/main/java/com/sundial/v1001/service/TwilightService.kt]: Updated the visibility modifiers per the suggestions here: https://kotlinlang.org/docs/visibility-modifiers.html

File: app/src/main/java/com/sundial/v1001/service/TwilightService.kt [hunk 4]
```diff
@@ -9,10 +9,10 @@ import kotlinx.coroutines.withContext
...
 }
-class TwilightService : ITwilightService{
-    override suspend fun fetchData() : List<Twilight>?{
+class TwilightService: ITwilightService{
+    internal fun fetchData(): List<Twilight>?{
```
Comment (Emiller321) [inline app/src/main/java/com/sundial/v1001/service/TwilightService.kt]: Updated the visibility modifiers per the suggestions here: https://kotlinlang.org/docs/visibility-modifiers.html

File: app/src/main/java/com/sundial/v1001/AppModule.kt [hunk 5]
```diff
@@ -6,6 +6,6 @@ import org.koin.androidx.viewmodel.dsl.viewModel
 import org.koin.dsl.module
 
 val appModule = module {
-    viewModel { MainViewModel(get()) }
-    single <ITwilightService>{ TwilightService() }
+    viewModel {MainViewModel(get())}
```
Comment (Emiller321) [inline app/src/main/java/com/sundial/v1001/AppModule.kt]: Eliminated unnecessary Whitespace per https://kotlinlang.org/docs/coding-conventions.html

File: app/src/main/java/com/sundial/v1001/dto/Twilight.kt [hunk 6]
```diff
@@ -3,6 +3,4 @@ package com.sundial.v1001.dto
 import java.util.Date
 import com.google.gson.annotations.SerializedName
 
-data class Twilight (@SerializedName("sunrise") var sunrise : Date, @SerializedName("sunset") var sunset : Date, ){
-
-}
\ No newline at end of file
+data class Twilight (@SerializedName("sunrise") var sunrise : Date, @SerializedName("sunset") var sunset : Date)
```
Comment (Emiller321) [inline app/src/main/java/com/sundial/v1001/dto/Twilight.kt]: Eliminated unnecessary whitespace including deleting a comma that was unnecessary per https://kotlinlang.org/docs/coding-conventions.html

Comment (discospiff) [reply app/src/main/java/com/sundial/v1001/dto/Twilight.kt]: Trailing commas are sometimes recommended.

I'd keep the curlies as well.

**Comment by discospiff** [comment]
From this review, I'd keep the internal modifier. 

However, I would not merge this branch, as there are a few cases where internal is added and suspend is removed.  These are two different concepts that are not mutually exclusive (access modifier and ability to save the state of the thread).
